tools/percolator/msgf2pin.xml

```diff
@@ -1,22 +1,56 @@
-<tool id="msgfplus2pin" name="MS-GF+ output to Pin converter" version="3.0">
+<tool id="msgfplus2pin" name="MS-GF+ output to Pin converter" version="3.0.1">
     <description>to create Percolator input files</description>
     <requirements>
         <requirement type="package" version="3.0">percolator</requirement>
     </requirements>
     <stdio>
         <exit_code range="1:"/>
     </stdio>
-    <command>
+    <command><![CDATA[
+             mkdir target && mkdir decoy &&
+             
+             #if $mzids.inputtype == "multi"
+             #for $key in $mzids.target.keys()
+             ln -s "$mzids.target[$key]" "target/$key" &&
+             echo "target/$key" >> targetmeta &&
+             #end for
+             #for $key in $mzids.decoy.keys()
+             ln -s "$mzids.decoy[$key]" "decoy/$key" &&
+             echo "decoy/$key" >> decoymeta &&
+             #end for
+             msgf2pin -o $pinout -e $enzyme
+             #if $matches
+             -m $matches
+             #end if
+             $aafreq $ptm $pngase $isotope targetmeta decoymeta
+             
+             #else
+             ln -s '$mzids.target' target/'$target.display_name' &&
+             ln -s '$mzids.decoy' decoy/'$decoy.display_name' &&
              msgf2pin -o $pinout -e $enzyme
              #if $matches
              -m $matches
              #end if
              $aafreq $ptm $pngase $isotope
-             "$target" "$decoy"
-    </command>
+             target/"$mzids.target.display_name" decoy/"$mzids.decoy.display_name"
+             #end if
+             
+    ]]></command>
     <inputs>
-        <param name="target" type="data" format="mzid,meta2pin" label="Target MSGF+ results or metafile pointing to them" />
-        <param name="decoy" type="data" format="mzid,meta2pin" label="Decoy MSGF+ results or metafile pointing to them" />
+        <conditional name="mzids">
+            <param name="inputtype" type="select" display="radio">
+                <option value="single">Single mzIdentML</option>
+                <option value="multi">Multiple mzIdentML metafile collection</option>
```

Comment:
I'm not sure I understand what you mean with metafile? is this just a dataset collection?